src/problem.jl

```diff
@@ -0,0 +1,147 @@
+abstract type AbstractProblem end
+
+"""
+    compute(problem, grid::Union{Grid,GridRSP})
+
+Compute problem `o` on a grid.
+"""
+function compute end
+
+"""
+    assess(p::AbstractProblem, g)
+
+Assess the memory and compute requirements of problem
+`p` on grid `g`. This can be used to indicate memory
+and time reequiremtents on a cluster
+"""
+function assess end
+
+"""
+    SolverMode
+
+Abstract supertype for ConScape solver modes.
+"""
+abstract type SolverMode end
+"""
+   MaterializedSolve()
+
+Compute all operations on a fully materialised Z matrix.
+
+This is inneficient for CPUS, but may be best for GPUs
+using CuSSP.jl ?
+"""
+struct MatrixSolve <: SolverMode end
+"""
+   ColumnSolve()
+
+Compute all operations column by column,
+after precomputing lu decompositions.
+
+TODO we can put LinearSolve.jl solver objects in the struct
+"""
+struct VectorSolve <: SolverMode end
+
+"""
+    Problem(ops...; mode, θ)
+
+Combine multiple compute operations into a single object, 
+to be run in the same job.
+"""
+@kwdef struct Problem{O,M<:SolverMode,T} <: AbstractProblem
+    ops::O
+    mode::M=VectorSolve()
+    θ::T=1.0
```

Comment:
See my comment above, I think that the Problem object should be more general and have a distance fields instead of `theta`.

Reply:
Yeah, if we factor distance function to objects we can just put theta in as a parameter where it is needed